src/java.base/aix/classes/sun/nio/fs/AixFileSystemProvider.java

```diff
@@ -52,15 +52,24 @@ AixFileStore getFileStore(UnixPath path) throws IOException {
         return new AixFileStore(path);
     }
 
+    static boolean supportsUserDefinedFileAttributeView(Path obj) {
+        try {
+            FileStore store = Files.getFileStore(obj);
+            return store.supportsFileAttributeView(UserDefinedFileAttributeView.class);
+        } catch (IOException e) {
+            return false;
+        }
+    }
+
     @Override
     @SuppressWarnings("unchecked")
     public <V extends FileAttributeView> V getFileAttributeView(Path obj,
                                                                 Class<V> type,
                                                                 LinkOption... options)
     {
         if (type == UserDefinedFileAttributeView.class) {
-            return (V) new AixUserDefinedFileAttributeView(UnixPath.toUnixPath(obj),
-                    Util.followLinks(options));
+            return !supportsUserDefinedFileAttributeView(obj) ? null :
+                (V) new AixUserDefinedFileAttributeView(UnixPath.toUnixPath(obj), Util.followLinks(options));
```

Comment:
I think should this be inverted, meaning use supportsUserDefinedFileAttributeView instead of ! supportsUserDefinedFileAttributeView and it will be clear to read.